### ISSUE TITLE
Pipeline freeze possibly due to Buildkit calling send on gRPC stream in parallel

### ISSUE THREAD

**Issue by sipsma** [dagger/dagger#5294]
There is a lot more context in this whole thread: https://github.com/dagger/dagger/pull/5229

tl;dr
* Long-standing issue with buildkit where certain grpc operations just freeze indefinitely until client disconnects.
* Reported a few years ago, but issue went stale https://github.com/grpc/grpc-go/issues/4722
* We started hitting this in Dagger, [here](https://github.com/dagger/dagger/pull/5072) and [here](https://github.com/dagger/dagger/pull/5229)
* Newer issue in grpc repo suggests that this is actually a bug on the user's part https://github.com/grpc/grpc-go/issues/5393#issuecomment-1148851417 (in this case, the user is buildkit)
* My unconfirmed theory on how to fix it upstream in Buildkit is [here](https://github.com/dagger/dagger/pull/5229#issuecomment-1577749213)

Given this has already bitten us twice (and been time consuming to debug both times), I think it's worth prioritizing a fix upstream.

I actually was able to trigger this in buildkit directly previously in the test case in this PR: https://github.com/moby/buildkit/pull/3832

But we should also verify that any fix we send upstream also fixes it in the case where we could repro easily (https://github.com/dagger/dagger/pull/5229)

**Comment by sipsma** [comment]
@matiasinsaurralde is looking into this (can't assign the issue to him for some reason)

**Comment by matiasinsaurralde** [comment]
Potential fix submitted to `buildkit`: https://github.com/moby/buildkit/pull/3940

**Comment by matiasinsaurralde** [comment]
Commit 1c56c7e0b5ea40b69445a31fcfb4356f9a2553e5 bumps `buildkit` dependency in Dagger to include the above fix 👍
